### ISSUE TITLE
Error has been raised. Exception: System.ArgumentNullException

### ISSUE THREAD

**Issue by dmitriy-shleht** [NLog/NLog#2630]
Hi!
**NLog version**: 4.4.13

**Platform**: .Net 4.0

**Current NLog config**

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" internalLogLevel="Trace" internalLogFile="C:\logs\nlog_cb.txt">

  <extensions>
    <add assembly="BL.ClientBank" />
  </extensions>
  
  <targets>
    <target xsi:type="LogReceiverSecureService"
      name="NlogService"
      endpointConfigurationName="ILogReceiver"
      endpointAddress="https://test-services.cbpmr.net:11443/client-bank/LR.svc"
      useBinaryEncoding="true"
      includeEventProperties="true">
    </target>
  </targets>

  <rules>
    <logger name="*" writeTo="NlogService" />
  </rules>
</nlog>
```
```cs
[Target("LogReceiverSecureService")]
public class LogReceiverSecureService : LogReceiverWebServiceTarget
{
    protected override IWcfLogReceiverClient CreateLogReceiver()
    {
        var client = base.CreateLogReceiver();
        client.ClientCredentials.UserName.UserName = "admin";
        client.ClientCredentials.UserName.Password = "admin";
        UseOneWayContract = true;
        return client;
    }
}
```
**Working:**

```cs
Logger _logger = LogManager.GetCurrentClassLogger();
_logger.Log(LogLevel.Info, "test");
```

**Not working:**

```cs
Logger _logger = LogManager.GetCurrentClassLogger();
var inf = new LogEventInfo
{
     Message = "test",
     Level = LogLevel.Info
};
_logger.Log(inf);
```

**Logs:**
[nlog_cb.txt](https://github.com/NLog/NLog/files/1850505/nlog_cb.txt)

**Comment by 304NotModified** [comment]
Thanks for the report. Stacktrace:

```
2018-03-27 08:13:32.2489 Error Error has been raised. Exception: System.ArgumentNullException: The value can not be undefined.
Parameter name: key
   в System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   в System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   в NLog.Targets.LogReceiverWebServiceTarget.AddValueAndGetStringOrdinal(NLogEvents context, Dictionary`2 stringTable, String value)
   в NLog.Targets.LogReceiverWebServiceTarget.TranslateEvent(LogEventInfo eventInfo, NLogEvents context, Dictionary`2 stringTable)
   в NLog.Targets.LogReceiverWebServiceTarget.TranslateLogEvents(IList`1 logEvents)
   в NLog.Targets.LogReceiverWebServiceTarget.Write(IList`1 logEvents)
   в NLog.Targets.Target.WriteAsyncThreadSafe(AsyncLogEventInfo logEvent)
```

**Comment by dmitriy-shleht** [comment]
version 4.5.1 will run on Framework 4?

**Comment by 304NotModified** [comment]
yes, it will be NLog 4.5.1 and NLog 4.x has support for the following platforms:



- .NET Framework 3.5, 4, 4.5, 4.6 & 4.7
- .NET Framework 4 client profile
- Xamarin Android
- Xamarin iOs
- Windows Phone 8
- Silverlight 4 and 5
- Mono 4
- ASP.NET 4 (NLog.Web package)
- ASP.NET Core (NLog.Web.AspNetCore package)
- .NET Core (NLog.Extensions.Logging package)
- .NET Standard 1.x - NLog 4.5
- .NET Standard 2.x - NLog 4.5
- UWP - NLog 4.5

See http://nlog-project.org/, header Support

**Comment by 304NotModified** [comment]
Found out that one of the values is probably `null` and that would result in this crash. Fix: #2638
